src/elitefurretai/utils/team_repo.py

```diff
@@ -13,14 +14,64 @@ class TeamRepo:
 
     def __init__(self, filepath: str = "data/teams"):
         self._teams: Dict[str, str] = {}
-        files = [
-            os.path.join(filepath, f)
-            for f in os.listdir(filepath)
-            if os.path.isfile(os.path.join(filepath, f))
-        ]
-        for file in files:
-            with open(file, "r") as f:
-                self._teams[os.path.basename(file).replace(".txt", "")] = f.read()
+
+        for format_folder in os.listdir(filepath):
+            format_name = format_folder
+            folder_path = os.path.join(filepath, format_folder)
+            if os.path.isdir(folder_path):
+                print(f"format name: {format_name}")
+                filenames = os.listdir(folder_path)
+                for filename in filenames:
+                    team_path = os.path.join(filepath, format_folder, filename)
+                    print(team_path)
+                    with open(team_path, "r") as f:
+                        team_string = f.read()
+                    self._teams[
+                        format_name + "_" + os.path.basename(team_path).replace(".txt", "")
+                    ] = team_string
+                    self.validate_team(team_string, format_name)
+
+    # Returns True if valid
+    def validate_team(
+        self, team: str, format: str, SHOWDOWN_PATH: str = "../pokemon-showdown"
+    ) -> bool:
+
+        output = True
+        ELITEFURRETAI_PATH = os.getcwd()
+        os.chdir(SHOWDOWN_PATH)
+
+        try:
+            result = subprocess.run(
+                ["./pokemon-showdown", "pack-team"],
+                input=team,
+                text=True,
+                capture_output=True,
+                check=True,
+            )
+            packed_team = result.stdout.strip()
+        except subprocess.CalledProcessError as e:
+            print(f"Error packing team: {e}")
+            print(f"Error output: {e.stderr}")
+            output = False
+
+        try:
+            result = subprocess.run(
+                ["./pokemon-showdown", "validate-team", format],
+                input=packed_team,
+                text=True,
+                capture_output=True,
+                check=True,
+            )
+            print(result.stdout)
+        except subprocess.CalledProcessError as e:
+            print(f"Error validating team: {e}")
+            print(f"Format: {format}")
+            print(f"Team (packed): {packed_team}")
+            print(f"Error output: {e.stderr}")
+            output = False
+
+        os.chdir(ELITEFURRETAI_PATH)
+        return output
 
     @property
     def teams(self) -> Dict[str, str]:
```

Comment:
One thing I was wondering -- do you think it would be good to add a `get` method to this?
1. So you could pass it a format and a name (`tr.get(format="", name="", validate=False)`), and it would retrieve that team, and optionally ensure it's valid?
2. You could similarly do a `get_formats()` that would return all formats we have for teams
3. You could similarly do `get_all(format="", validate=False)` that would get all teams of a format?

This would allow us to abstract all knowledge of how we store the teams internally, and would allow users to just need to know these three common sense functions. What do you think?

Reply:
This makes a lot of sense and would definitely clean things up. 
Will go ahead and implement them (I think it'd also be good to add a `verbose` parameter for `get` methods that return teams).